docs/cli/settings.md

```diff
@@ -18,58 +18,61 @@ Note: Workspace settings override user settings.
 Here is a list of all the available settings, grouped by category and ordered as
 they appear in the UI.
 
+<!-- SETTINGS-AUTOGEN:START -->
+
 ### General
 
-| UI Label                        | Setting                            | Description                                                                  | Default     |
-| ------------------------------- | ---------------------------------- | ---------------------------------------------------------------------------- | ----------- |
-| Preview Features (e.g., models) | `general.previewFeatures`          | Enable preview features (e.g., preview models).                              | `false`     |
-| Vim Mode                        | `general.vimMode`                  | Enable Vim keybindings.                                                      | `false`     |
-| Disable Auto Update             | `general.disableAutoUpdate`        | Disable automatic updates.                                                   | `false`     |
-| Enable Prompt Completion        | `general.enablePromptCompletion`   | Enable AI-powered prompt completion suggestions while typing.                | `false`     |
-| Debug Keystroke Logging         | `general.debugKeystrokeLogging`    | Enable debug logging of keystrokes to the console.                           | `false`     |
-| Session Retention               | `general.sessionRetention`         | Settings for automatic session cleanup. This feature is disabled by default. | `undefined` |
```

Comment:
Session Retention removed from the list as it isn't shown in the settings dialog.